plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -0,0 +1,234 @@
+        );
+        this._moduleVersion = moduleVersion;
+        return moduleExports;
+      },
+      (moduleExports, moduleVersion) => {
+        api.diag.debug(
+          `Removing patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+      }
+    );
+
+    module.files.push(
+      this.getClientNodeModuleFileInstrumentation('src'),
+      this.getClientNodeModuleFileInstrumentation('lib'),
+      this.getRunnerNodeModuleFileInstrumentation('src'),
+      this.getRunnerNodeModuleFileInstrumentation('lib'),
+      this.getRunnerNodeModuleFileInstrumentation('lib/execution')
+    );
+
+    return module;
+  }
+
+  private getRunnerNodeModuleFileInstrumentation(basePath: string) {
+    return new InstrumentationNodeModuleFile<typeof knex>(
+      `knex/${basePath}/runner.js`,
+      constants.SUPPORTED_VERSIONS,
+      (Runner: any, moduleVersion) => {
+        api.diag.debug(
+          `Applying runner.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Runner.prototype,
+          'query',
+          this.wrapQuery.bind(this)
+        );
+        return Runner;
+      },
+      (Runner: any, moduleVersion) => {
+        api.diag.debug(
+          `Removing patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this._unwrap(Runner.prototype, 'query');
+        return Runner;
+      }
+    );
+  }
+
+  private getClientNodeModuleFileInstrumentation(basePath: string) {
+    return new InstrumentationNodeModuleFile<typeof knex>(
+      `knex/${basePath}/client.js`,
+      constants.SUPPORTED_VERSIONS,
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Applying client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'queryBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'schemaBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'raw',
+          this.storeContext.bind(this)
+        );
+        return Client;
+      },
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Removing client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this._unwrap(Client.prototype, 'queryBuilder');
+        this._unwrap(Client.prototype, 'schemaBuilder');
+        this._unwrap(Client.prototype, 'raw');
+        return Client;
+      }
+    );
+  }
+
+  private wrapQuery(original: Function) {
+    const instrumentation = this;
+    return function wrapped_logging_method(this: any, query: any) {
+      const config = this.client.config;
+
+      const maxLen = (instrumentation._config as types.Config).maxQueryLength!;
+      const attributes: any = {
```

Comment:
you can type it with [`SpanAttributes`](https://github.com/open-telemetry/opentelemetry-js-api/blob/main/src/trace/attributes.ts#L17) instead of any